obspy/io/rg16/tests/test_util.py

```diff
@@ -0,0 +1,106 @@
+"""
+tests for the Utilities of rg16
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA
+
+import unittest
+from io import BytesIO
+
+from obspy.io.rg16.util import _read
+
+
+def byte_io(byte_str):
+    """
+    write byte_str to BytesIO object, return.
+    """
+    return BytesIO(byte_str)
+
+
+class TestRG16Util(unittest.TestCase):
+    """
+    Tests for the read function, which should read all the weird binary
+    formats used in rg16.
+    """
+
+    # --- tests bcd
+
+    bcd = [
+        (b'\x99', 1, 99),
+        (b'\x99\x01', 2, 9901),
+    ]
+
+    def test_read_bcd(self):
+        """
+        Ensure bcd encoding returns expected values.
+        """
+        for byte, length, answer in self.bcd:
+            out = _read(byte_io(byte), 0, length, 'bcd')
+            self.assertEqual(out, answer)
+
+    def test_ff_raises(self):
+        """
+        Ensure FF raises. BCD values for any half byte past 9 should raise.
+        """
+        with self.assertRaises(ValueError) as e:
+            _read(byte_io(b'\xFF'), 0, 1, 'bcd')
+        assert 'invalid bcd values' in str(e.exception)
+
+    # --- test half byte reads
+
+    halfsies = [
+        (b'\x45', '>i.', 4),
+        (b'\x45', '<i.', 5),
+        (b'\xfa', '>i.', 15),
+        (b'\xfa', '<i.', 10),
+    ]
+
+    def test_read_half_bit(self):
+        """
+        Ensure reading half bytes (4 bit) works.
+        """
+        for byte, format, answer in self.halfsies:
+            self.assertEqual(_read(byte_io(byte), 0, 1, format), answer)
+
+    # --- test 24 bit (3 byte) reads
+
+    why_use_3_bytes = [  # seriously, how expensive is one extra byte!?
```

Comment:
Hehe :-) One of the format (forgot which one) in ObsPy has some kind of 6 byte number representation...